common/types_test.go

```diff
@@ -37,3 +41,93 @@ func TestDataVersion(t *testing.T) {
 	require.Equal(t, ForkVersionStringCapella, consensusspec.DataVersionCapella.String())
 	require.Equal(t, ForkVersionStringDeneb, consensusspec.DataVersionDeneb.String())
 }
+
+func TestEncodeDecodeTxs(t *testing.T) {
+	testAddr := common2.HexToAddress("0xB9D7a3554F221B34f49d7d3C61375E603aFb699e")
+
+	tx1 := gethtypes.NewTx(&gethtypes.LegacyTx{
+		Nonce:    1,
+		GasPrice: big.NewInt(1),
+		Gas:      1,
+		To:       &testAddr,
+		Value:    big.NewInt(1),
+		Data:     []byte("tx1"),
+	})
+	tx2 := gethtypes.NewTx(&gethtypes.LegacyTx{
+		Nonce:    2,
+		GasPrice: big.NewInt(2),
+		Gas:      2,
+		To:       &testAddr,
+		Value:    big.NewInt(2),
+		Data:     []byte("tx2"),
+	})
+
+	txs := encodeTransactions([]*gethtypes.Transaction{tx1, tx2})
+
+	decodedTxs, err := DecodeTransactions(txs)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(decodedTxs))
+	decodedTx1 := decodedTxs[0]
+	decodedTx2 := decodedTxs[1]
+	require.Equal(t, tx1.Nonce(), decodedTx1.Nonce())
```

Comment:
there is probably a deep equal thing somewhere to compare all these, no?

Reply:
yes will explore that. but I think the things I check for should suffice for now I believe.